packages/contracts-bedrock/scripts/DeployConfig.s.sol

```diff
@@ -161,6 +175,17 @@ contract DeployConfig is Script {
         useInterop = _readOr(_json, "$.useInterop", false);
     }
 
+    function latestGenesisFork() public view returns (Fork) {
```

Comment:
Thoughts on using `Config.Fork` here and update this function to be `fork()` and then prioritize the deploy config, if not present fall back to the env var?

Reply:
I thought the `DeployConfig` should be the sole abstraction around the deploy config json file, whereas the `Config` is for reading env vars etc. So as to not mix abstractions I preferred to handle the fallback logic in the L2 genesis script. But I can see that it's nicer to encapsulate the config loading logic into one of these instead of at the consumer end, in the L2 genesis script.

Wondering if we want to make one or the other the higher abstraction. E.g. we could make `Config` the higher abstraction that accesses `DeployConfig` in its `Config.fork()` function. For this I'd just need to move the two L2 genesis enums `OutputMode` and `Fork` into their own solidity file, like `scripts/GenesisConfig.s.sol`.

Reply:
I definitely see your thinking regarding keeping the abstractions thin, from a consumer pov I do think its nice to have a single place to read config relevant to deployment, which to me is the `DeployConfig`. We don't really have this pattern of env vars in the `DeployConfig` today, but i do think that we want a single source of truth where the fork is pulled from which I think that should be the `DeployConfig` and we can start wrapping `Config` with the `DeployConfig` for the deploy config -> env var fallback, so that devs can always just use `DeployConfig` for the source of ultimate truth, enabling a consistent way that files and env vars relate to each other with the deploy config, ie which takes precedence

Reply:
That also makes sense, added `fork()` to `DeployConfig`, which internally still calls `Config.fork()` and if this is unset, reads the latest genesis fork from the deploy config.